src/main/scala/org/broadinstitute/dsde/firecloud/utils/StreamingPassthrough.scala

```diff
@@ -90,10 +91,18 @@ trait StreamingPassthrough
     // Convert the URI to the one suitable for the remote system
     val targetUri = convertToRemoteUri(req.uri, localBasePath, remoteBaseUri)
 
-    // Remove unwanted headers. Akka automatically adds a Timeout-Access to the request
+    // Remove unwanted headers:
+    // Timeout-Access: Akka automatically adds a Timeout-Access to the request
     // for internal use in managing timeouts; see akka.http.server.request-timeout.
     // This header is not legal to pass to an external remote system.
-    val targetHeaders = req.headers.filter(_.isNot(`Timeout-Access`.lowercaseName))
+    //
+    // Host: The Nginx ingress controller used in Terra's BEE environments uses the Host header for routing,
+    // so we remove and set it with targetUri host
+    val filteredHeaders = req.headers.
+      filter(_.isNot(`Timeout-Access`.lowercaseName)).
+      filter(_.isNot(Host.lowercaseName))
```

Comment:
I suggest consolidating this into one `filter` operation which checks both conditions:
```scala
    val filteredHeaders = req.headers.filter { hdr =>
      hdr.isNot(`Timeout-Access`.lowercaseName) &&
        hdr.isNot(Host.lowercaseName)
    }
```

Reply:
integrated this, thanks!